src/windows/main_window.py

```diff
@@ -2058,17 +2058,27 @@ def removeSelection(self, item_id, item_type):
     def save_settings(self):
         s = settings.get_settings()
 
-        # Save window state and geometry (saves toolbar and dock locations)
+        # Save window state and geometry (saves toolbar, dock locations and frozen UI state)
         s.set('window_state_v2', qt_types.bytes_to_str(self.saveState()))
         s.set('window_geometry_v2', qt_types.bytes_to_str(self.saveGeometry()))
+        if self.actionUn_Freeze_View.isVisible():
+            docks_frozen = 'True'
+        else:
+            docks_frozen = 'False'
+        s.set('docks_frozen', docks_frozen)
 
     # Get window settings from setting store
     def load_settings(self):
         s = settings.get_settings()
 
-        # Window state and geometry (also toolbar and dock locations)
+        # Window state and geometry (also toolbar, dock locations and frozen UI state)
         if s.get('window_state_v2'): self.restoreState(qt_types.str_to_bytes(s.get('window_state_v2')))
         if s.get('window_geometry_v2'): self.restoreGeometry(qt_types.str_to_bytes(s.get('window_geometry_v2')))
+        if s.get('docks_frozen') == 'True':
```

Comment:
By the same token, this should just be `if s.get('docks_frozen'):` since it'll be stored as a boolean true/false value and you can just test it directly. (If the setting is missing, that'll still imply false, which is what we'd want.)

Reply:
Storing non-string values in Settings is usually bad practice. The conversion was made intentionally.

Reply:
JSON supports boolean values natively, and the OpenShot convention is to use them:
https://github.com/OpenShot/openshot-qt/blob/e48cc2be1865110a0a8491ff4a5f887994650aa8/src/settings/_default.settings#L464-L470

Reply:
> When later you decide to change the UI, for example, if you'll try to replace the boolean checkbox, with the list `0, 1, 2, 3, ... n` you'll be unable to reuse the existing setting (variable) that was saved on users PCs with the previous build of OpenShot.

True, and for all those reasons, we wouldn't want to reuse it. If a situation like that occurs (which it has, in the past), we simply version the setting. `docks_frozen_v2` can be a string, or a list of dock names, or whatever we choose, and (if it makes sense) the old value is migrated from the user's existing settings when there's no value for the new preference. That's already been done for one or two preferences that have changed semantics in the past.

Doing it that way, the user's saved settings remain compatible in both directions (for newer _and_ older versions), since we haven't changed the meaning of the older version's preference, by setting a value that it doesn't know how to interpret.

Reply:
Really, who cares, user should reconfigure UI because it was programmed in a new way...

Reply:
Well... I care, for one. I often run multiple different versions of OpenShot for debugging and troubleshooting purposes. So I don't want to have to worry about my preferences file breaking an older version.

Reply:
Yeah, if we are correctly storing the boolean value, then `if s.get('docks_frozen')` is pretty elegant. :+1: If for some reason this changes in the future to a more complex type, we would certainly change the "key" and handle the upgrade path from old setting to new setting. But this is a pretty minor issue, so let's just agree to disagree, and stick with the boolean value for now. That gets my vote, haha.

Reply:
Otherwise, I like this! LG!

Reply:
> ...So I don't want to have to worry about my preferences file breaking an older version.

@ferdnyc It never would, if you always reading and processing strings. This is the advantage of the method - forward and backward compatibility, unlike the different variable types.

Reply:
@SuslikV We seem to have reached a point where you can either win this argument, _or_ you can get your PR merged. Your choice.

Reply:
The `QDockWidget::AllDockWidgetFeatures` or _QDockWidget.AllDockWidgetFeatures_ was deprecated, in favor of using individual flags. Thus I strongly recommend to use strings here. I'll try to rewrite the code one more time.